File: 2. Even Fibonacci Numbers/solution.ts
function fibonacciSum(limit: number): number {
    let fibSum: number = 0;
    let a: number = 1;
    let b: number = 2;

    while (a <= limit) {
        if (a % 2 === 0) {
            fibSum += a;
        }
        [a, b] = [b, a + b];
    }

    return fibSum;
}

const limit: number = 4000000;
const result: number = fibonacciSum(limit);
console.log(result)

File: 2. Even Fibonacci Numbers/solution.py
def fibonacci_sum(limit):
    fib_sum = 0
    a, b = 1, 2
    while a <= limit:
        if a % 2 == 0:
            fib_sum += a
        a, b = b, a + b
    return fib_sum

limit = 4000000
result = fibonacci_sum(limit)
print(result)